presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -456,9 +456,12 @@ public IndexMetadata getIndexMetadata(String index)
                     }
                     result.add(new IndexMetadata.Field(name, new IndexMetadata.DateTimeType(formats)));
                 }
-                else {
+                else if (!type.equals("object")) {
```

Comment:
I don't think this is correct. According to Elasticsearch docs (https://www.elastic.co/guide/en/elasticsearch/reference/current/object.html):

> You are not required to set the field type to object explicitly, as this is the default value.

This seems to imply that a mapping that contains fields can have an explicit `type = "object"`, which means it would be excluded. Instead, we should check whether the field has type `object` and *empty* properties to skip it.

Reply:
I agree with the way you understand documentation, however in elasticsearch mapping implementation, we have :
https://github.com/elastic/elasticsearch/blob/d1334b812226acbe48fdae84bbf841c484d6e675/server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java#L534

Reply:
Thanks for the pointers!

This whole section can now be simplified to make the checks more explicit and the flow easier to follow:
```java
String type = "object";
if (value.has("type")) {
    type = value.get("type").asText();
}

switch (type) {
    case "date":
        ...
        break;
    case "object":
        if (value.has("properties")) {
            result.add(new IndexMetadata.Field(name, parseType(value.get("properties"))));
        }
        else {
            LOG.debug("Ignoring empty object field: %s", name);
        }
        break;
    default:
        result.add(new IndexMetadata.Field(name, new IndexMetadata.PrimitiveType(type)));
}
```